vendor/engines/split_accounts/app/services/split_accounts/attribute_splitter.rb

```diff
@@ -0,0 +1,23 @@
+module SplitAccounts
+  class AttributeSplitter
+    attr_reader :splittable_attributes
+
+    def initialize(*splittable_attributes)
+      @splittable_attributes = splittable_attributes
+    end
+
+    def split(original_object, split_object, split)
+      splittable_attributes.each do |attr|
+        split_object.public_send "#{attr}=", floored_amount(split.percent, original_object.public_send(attr))
+      end
+    end
+
+    private
+
+    def floored_amount(percent, value)
+      return BigDecimal(0) if percent == 0 || value.blank?
+      amount = BigDecimal(value) * BigDecimal(percent) / 100
+      amount.floor(2)
+    end
+  end
+end
```

Comment:
I thought about how I could move `apply_remainder` over, and it was going to be tricky because it needs to know about a lot of the local objects, the original object, the list of split objects. 

Also, I'm not applying remainders to the reservation splits. The main reason I decided to forego that is because the splitting on minutes is a float (not rounded), and we're only reporting down to the tenth of an hour. It didn't seem worth it to deal with the minutes with remainders.